Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -177,7 +179,8 @@ SelectorChecker::SelectorChecker(Document& document)
 
 bool SelectorChecker::match(const CSSSelector& selector, const Element& element, CheckingContext& checkingContext) const
 {
-    LocalContext context(selector, element, checkingContext.resolvingMode == SelectorChecker::Mode::QueryingRules ? VisitedMatchType::Disabled : VisitedMatchType::Enabled, checkingContext.pseudoId);
+    auto pseudoElementIdentifier = checkingContext.pseudoId == PseudoId::None ? std::nullopt : std::optional(Style::PseudoElementIdentifier { checkingContext.pseudoId, checkingContext.pseudoElementNameArgument });
```

Comment:
```
std::optional<PseudoElementIdentifier> pseudoElementIdentifier;
if (checkingContext.pseudoId != None)
   pseudoElementIdenfitier = Style::PseudoElementIdentifier { checkigContext.pseudoId, checkContext.pseudoElemet };
```

might be clearer ?

Reply:
I like the one liner better, and the ternary should go away once I figure out how to switch `CheckingContext` and CSSJIT to use PseudoElementIdentifier